### ISSUE TITLE
Prevent duplicate importation from external linked origins.

### ISSUE THREAD

**Issue by rokejulianlockhart** [bookwyrm-social/bookwyrm#2663]
**Is your feature request related to a problem? Please describe.**
If you search for Houseki no Kuni, the first installment of the series appears in the search result from OpenLibrary. If a user imports that novel, the server does not prevent that user nor any other (not that the distinction is important in this case) importing it again.

**Describe the solution you'd like**
Record when a piece of literature is imported from an external, connected source, and consequently prevent subsequent importation of that piece of literature.

<--
**Describe alternatives you've considered**

**Additional context**
--!>

**Comment by mouse-reeve** [comment]
When a book is imported, it is matched against unique identifiers, and it isn't re-imported if a match is found. So if a book is imported from OpenLibrary, a subsequent import from OpenLibrary should match on the openlibrary_key field and should not create a duplicate entry.

There are still sources of duplicates -- books are imported from different sources which don't have a shared key, duplicates exist in the sources themselves, or a user manually creates a book. But as far as I know re-importing from the same source should not cause any data consistency issues. Do you have a specific case where it does create two entries when importing from the same source?

**Comment by rokejulianlockhart** [comment]
Yes, @mouse-reeve: An installment of Houseki no Kuni from OpenLibrary. I imported the novel, modified some basic metadata to be consistent with the already existent other installment of the series, and accidentally imported the same novel from OL again. It successfully imported.

I certainly did not manually remove the `openlibrary_key`.

**Comment by mouse-reeve** [comment]
Thank you for the extra details - when it imported again, it created a new, second copy of the book?

**Comment by rokejulianlockhart** [comment]
@mouse-reeve, from what I recall, yes. However, by now, my memory is becoming spotty.
